### ISSUE TITLE
Admin:Curation and Collections:Contribute permissions need to work with the Collection Add-ons Create endpoint

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-server#8747]
Collection Add-ons Edit and Collection Add-ons Delete APIs are both configured to work with the new Admin:Curation and Collections:Contribute permissions, but it appears that Collection Add-ons Create has not. It needs to also support those permissions so that appropriate users can add a new add-on to a collection.

**Comment by bobsilverberg** [comment]
Note that this blocks https://github.com/mozilla/addons-frontend/issues/5163 which is slated to land on prod next push, so should be treated as a fairly high priority.

**Comment by bobsilverberg** [comment]
@eviljeff All of the UI to support the `Admin:Curation` and `Collections:Contribute` permissions have either landed, or will land this week and next week. Unfortunately but we still have this issue that users with these permissions cannot add an add-on to a collection, which manifests as a bug in the app. If a user tries to add an add-on we get a "Not found" error back from the API.

Do you think it will be possible to address this in the near future?

**Comment by eviljeff** [comment]
@diox implemented these new permissions - do you know what's missing here?

**Comment by diox** [comment]
This is a limitation of how the permissions were implemented. For both `AllowCollectionContributor` and `AllowContentCurators` we disallow the `create` action on the `CollectionViewSet`: The former is irrelevant in that context, they don't need to create a new collection, and the latter is more of a technical issue, we don't have access to the user object in the permission to make sure the collection is going to be created on the right user.

The fact that it affects `CollectionAddonViewSet` and therefore adding add-ons to a collection is a bug. We need to rethink this a bit...

**Comment by diox** [comment]
This is difficult to follow and I may be wrong but I... no longer think there is an addons-server issue here. In `CollectionAddonViewSet`, we don't do through normal permission checks, instead we get the object from the `CollectionViewSet`, and that's what performs the permission checks.

I originally thought that by doing this, the `PreventActionPermission` part of the permissions would prevent some actions to take place... but this is not the case, because when we instantiate the `CollectionViewSet`, we never set an action, so that part always succeeds, never prevents the action, so only the other part of the permission check is used, which is `AllowCollectionContributor`, or `AllowContentCurators`, or `AllowReadOnlyIfPublic`. We have tests proving that it does work correctly.

**TL;DR:** although the code to handle this is confusing and kinda flimsy, I don't think there is an issue on the addons-server side. It seems to be working just fine.

**Comment by bobsilverberg** [comment]
My apologies @diox, it was indeed a bug on the front end, which I assumed was a server bug because I was getting an error back from the API, but the API was being called incorrectly.
